process/coordinator/sovereignChainProcess.go

```diff
@@ -168,3 +168,42 @@ func (sctc *sovereignChainTransactionCoordinator) CreateMbsAndProcessCrossShardT
 
 	return createMBDestMeExecutionInfo.miniBlocks, createMBDestMeExecutionInfo.numTxAdded, allMBsProcessed, nil
 }
+
+// CreateMbsAndProcessTransactionsFromMe should only create miniblocks and not process transactions
+func (sctc *sovereignChainTransactionCoordinator) CreateMbsAndProcessTransactionsFromMe(
```

Comment:
I don't think that this overwritten method is really necessary, as the only difference between this one and the main implementation is the following code: 
```
	interMBs := tc.CreatePostProcessMiniBlocks()
	if len(interMBs) > 0 {
		miniBlocks = append(miniBlocks, interMBs...)
	}

```
Which should always return an empty slice, as there is no execution done by the proposer at the time of block creation. Anyway, should be tested, but I think it is better to extract only the above code in separated methods for main and for sovereign, and keep the most of the common code only in one place (mainchain).

Reply:
Right, removed it, tested it and it worked